### PR TITLE
[Core][Proposal] Adding pSetGeometry to GeometricalObject

### PR DESCRIPTION
**Pull request by loumalouomega**
I was trying to use the coupled geometry with the contact app. I was unable to update the geometry in the condition once this is already created. This is one option I thought, let me know if you agree

### CODE REVIEW

**Comment by tteschemacher** [comment]
I had a similiar problem, once. Have you tried doing it like this?

condition->pGetGeometry() = p_new_geometry;

**Comment by loumalouomega** [comment]
> I had a similiar problem, once. Have you tried doing it like this?
> 
> condition->pGetGeometry() = p_new_geometry;

Does not work, as pGetGeometry is not referenced. I am trying to solve from my side, but it requires to dynamic_ _cast

**Comment by tteschemacher** [comment]
Ok. Anyways, I think it is a good idea changing it like this!

File: kratos/tests/cpp_tests/geometries/test_coupling_geometry.cpp [hunk 1]
```diff
@@ -95,6 +109,59 @@ namespace Kratos {
...
+            auto p_triangle_master = GenerateNodeTriangle();
+            auto p_triangle_slave = GenerateNodeTriangle2();
+
+            Geometry<NodeType>::Pointer p_coupling_geometry = Kratos::make_shared<CouplingGeometry<NodeType>>(p_triangle_master, p_triangle_slave);
+
+            // Creating a model and an element
+            Model current_model;
+            ModelPart& r_model_part = current_model.CreateModelPart("Main");
+            auto p_elem_prop = r_model_part.CreateNewProperties(1);
+            auto p_element = r_model_part.CreateNewElement("Element2D3N", 1, p_triangle_master->Points(), p_elem_prop); // Creating the element
```
Comment (tteschemacher) [inline kratos/tests/cpp_tests/geometries/test_coupling_geometry.cpp:126]: Should be possible to directly put the geometry. This comes with #4872

Comment (loumalouomega) [reply kratos/tests/cpp_tests/geometries/test_coupling_geometry.cpp:126]: The CreateNewElement should allow pointer to geometries directly IMO. I was thinking to define that, but this PR was shorter

**Comment by tteschemacher** [comment]
I would be for adding this feature. @philbucher @pooyan-dadvand is that ok for you, too?

**Comment by loumalouomega** [comment]
> I would be for adding this feature. @philbucher @pooyan-dadvand is that ok for you, too?

It is not needed anymore, but may be useful...

**Comment by tteschemacher** [comment]
I would like to use it also at another place.

**Comment by pooyan-dadvand** [comment]
I'm not sure if I understood well the problem. Do you want to change the geometry or update it?

With this method you are changing the geometry which condition is pointing to.

**Comment by tteschemacher** [comment]
Well, actually we want to change the geometry.

**Comment by loumalouomega** [comment]
> I'm not sure if I understood well the problem. Do you want to change the geometry or update it?
> 
> With this method you are changing the geometry which condition is pointing to.

 That's the idea

**Comment by pooyan-dadvand** [comment]
Added to @KratosMultiphysics/technical-committee project for approval

**Comment by loumalouomega** [comment]
I will do a new PR to merge with the master

**Pull request closed without merging**